pages/data-migration/best-practices.mdx

```diff
@@ -0,0 +1,514 @@
+and property being matched in Cypher queries. For example, in the following
+queries:
+
+```cypher
+CREATE (:Person {id: 1});
+CREATE (:Person {id: 2});
+
+MATCH (p1:Person {id: 1}), (p2:Person {id:2})
+CREATE (p1)-[:IS_FRIENDS_WITH]->(p2);
+```
+
+It makes sense to create a label-property index `:Person(id)` for faster
+matching and relationship creation. The best practice is to create and use an
+index on properties containing many distinct (ideally unique) integer values,
+such as identification numbers. Choosing the right data to create indexes on is
+important, as indexing all the content will not improve the database speed.
+Indexing won't bring any improvements on a low-cardinality property such as
+gender.
+
+Be aware that **creating a label-property index will not create a label index**.
+If there is a need to optimize queries that fetch nodes by label, create a label
+index, too. A transaction which creates index or constraint will prevent other
+transactions from running, because such transactions take a unique lock for the
+whole duration of the transaction.
+
+There are some downsides to indexing too. Each index requires extra storage
+(takes up RAM memory), which can greatly impact a large dataset. Besides that,
+indexes slow down write operations to the database because the structures in the
+index are dynamically updated on modifications or insertions of new nodes. That
+means you should avoid creating indexes on properties that are being updated
+frequently.
+
+<Cards>
+    <Card
+    title="Read more about indexing"
+    href="/fundamentals/indexes"
+    />
+</Cards>
+
+### Use in-memory analytical storage mode
+
+Memgraph supports three storage modes: `IN_MEMORY_TRANSACTIONAL`, `IN_MEMORY_ANALYTICAL` and `ON_DISK_TRANSACTIONAL`
+**To speed up import, use `IN_MEMORY_ANALYTICAL` storage mode**. 
+
+Data import is faster in `IN_MEMORY_ANALYTICAL` storage mode because `Delta`
+objects are disabled there, which ensures there is no memory overhead. The
+`Delta` objects track all changes, and that requires a lot of memory in case of
+frequent updates. Not having `Deltas` when importing data, speeds up the process
+significantly. 
+
+The `Delta` objects are crucial to ensure atomicity, consistency, isolation and
+durability. If you are follwing instuctions, your import process does not need to depend on ACID properties, 
+and it is safe to use `IN_MEMORY_ANALYTICAL` storage mode, even in concurrent import scenarios.  
+
+Once you’re done with the data import process, if you have write enabled workloads you should consider
+switching back to the `IN_MEMORY_TRANSACTIONAL` mode, which favors
+strongly-consistent ACID transactions. On a larger scale import, if you stick with the default
+`IN_MEMORY_TRANSACTIONAL` storage mode, you’ll experience slower import time and
+spend more resources during data import because of `Delta` objects. 
+
+TODO(@antejavor): Do we need a link to estimates in possible performance numbers? 
```

Comment:
```suggestion
```

Reply:
I would drop this, not to block the merge today, and linked numbers once we have them.